### PR TITLE
feat: revise test code for survey service and controller

### LINKED ISSUE #107: [P0] Revise test code for survey service and controller

### PR DESCRIPTION
**Pull request by bezzang-dev**
### Added 
- unit test code for logic that performs validation in the survey **service** layer.
- integrate test code in the survey **controller** layer.

### Modified
- the part where `StringUtil.trim()` function was not being used.

### FIXME
In `com/thesurvey/api/controller/SurveyControllerTest.java`,
I'm not sure why the `testDeleteSurvey()` function is not actually performing the deletion.
Does `MockMVC` actually perform the creation request but not the deletion request?

This PR closes #107 .

### CODE REVIEW

File: api/src/test/java/com/thesurvey/api/controller/SurveyControllerTest.java [hunk 1]
```diff
@@ -72,28 +90,123 @@ public class SurveyControllerTest extends BaseControllerTest {
...
     AuthenticationManager authenticationManager;
 
+    SurveyRequestDto surveyRequestDto;
+    QuestionRequestDto questionRequestDto;
+    QuestionOptionRequestDto questionOptionRequestDto;
+    JSONObject mockSurvey;
+
     @BeforeAll
-    void makeMockUser() throws Exception {
+    void setupBeforeAll() throws Exception {
```
Comment (seiwonpark) [inline api/src/test/java/com/thesurvey/api/controller/SurveyControllerTest.java:99]: Very nice!

File: api/src/test/java/com/thesurvey/api/service/SurveyServiceTest.java [hunk 2]
```diff
@@ -1,40 +1,239 @@
...
+            .build();
+
+        assertThrows(BadRequestExceptionMapper.class,
+            () -> surveyService.createSurvey(authentication, surveyRequestDto));
+    }
+
+    /**
+     * Test validate for when the start time of the survey is set after the end time.
+     */
+
```
Comment (seiwonpark) [inline api/src/test/java/com/thesurvey/api/service/SurveyServiceTest.java]: Could you remove this blank line here?

**Review by seiwonpark — APPROVED** [review]
LGTM. Thanks!

For `MockMvc`, I'll take a look at it.

File: api/src/test/java/com/thesurvey/api/service/SurveyServiceTest.java (hunk 2 above)
Comment (bezzang-dev) [reply api/src/test/java/com/thesurvey/api/service/SurveyServiceTest.java]: Okay!

**Pull request merged**